reflex/model.py

```diff
@@ -75,6 +93,154 @@ def get_db_engine():
+                )
+            return op
+
+        def run_autogenerate(rev, context):
+            revision_context.run_autogenerate(rev, context)
+            return []
+
+        with alembic.runtime.environment.EnvironmentContext(
+            config=config,
+            script=script_directory,
+            fn=run_autogenerate,
+        ) as env:
+            env.configure(
+                connection=connection,
+                target_metadata=sqlmodel.SQLModel.metadata,
+                render_item=cls._alembic_render_item,
+                process_revision_directives=writer,  # type: ignore
+            )
+            env.run_migrations()
+        changes_detected = False
+        if revision_context.generated_revisions:
+            upgrade_ops = revision_context.generated_revisions[-1].upgrade_ops
+            if upgrade_ops is not None:
+                changes_detected = bool(upgrade_ops.ops)
+        if changes_detected:
+            for _script in revision_context.generate_scripts():
+                pass  # must iterate to actually generate the scripts
+        return changes_detected
+
+    @classmethod
+    def _alembic_upgrade(
+        cls,
+        connection: sqlalchemy.engine.Connection,
+        to_rev: str = "head",
+    ) -> None:
+        """Apply alembic migrations up to the given revision.
+
+        Args:
+            connection: sqlalchemy connection to use when performing upgrade
+            to_rev: revision to migrate towards
+        """
+        config, script_directory = cls._alembic_config()
+
+        def run_upgrade(rev, context):
+            return script_directory._upgrade_revs(to_rev, rev)
+
+        # apply updates to database
+        with alembic.runtime.environment.EnvironmentContext(
+            config=config,
+            script=script_directory,
+            fn=run_upgrade,
+        ) as env:
+            env.configure(connection=connection)
+            env.run_migrations()
+
+    @classmethod
+    def automigrate(cls) -> Optional[bool]:
+        """Generate and execute migrations for all sqlmodel Model classes.
+
+        If alembic is not installed or has not been initialized for the project,
+        then no action is performed.
+
+        If models in the app have changed in incompatible ways that alembic
+        cannot automatically generate revisions for, the app may not be able to
+        start up until migration scripts have been corrected by hand.
+
+        Returns:
+            True - indicating the process was successful
+            None - indicating the process was skipped
+        """
+        if not has_alembic or not Path(constants.ALEMBIC_CONFIG).exists():
+            return
+
+        with cls.get_db_engine().connect() as connection:
+            cls._alembic_upgrade(connection=connection)
+            changes_detected = cls._alembic_autogenerate(connection=connection)
+            if changes_detected:
```

Comment:
Maybe we can give the user a prompt here to confirm if they want to migrate

Reply:
Or make a setting If the user wants to update automatically or ask

Reply:
@danik292 We have a follow up #1336 where we will add this as a config option